docs/user/testing.md

```diff
@@ -0,0 +1,151 @@
+# Testing
+
+In this section, we will talk about how to test different modules and algorithms in garage.
+
+In garage, we use [pytest](https://docs.pytest.org/en/stable/getting-started.html) to carry out tests.
+
+## Test modules
+
+All test files are stored under `garage/tests` in the garage respository.
```

Comment:
can you update all of these paths to remove the first `garage/` -- consider "root" to be inside the garage folder. so the path would be `tests/`

Reply:
to be more specific here:

`tests/` contains all tests and supporting code
`tests/garage` contains unit tests
`tests/integration_tests` contains integration tests
modules in `tests/` and `tests/fixtures` contain helper code and fixtures which make writing good tests easier.